llama_index/vector_stores/elasticsearch.py

```diff
@@ -465,3 +531,360 @@ def query(
         return VectorStoreQueryResult(
             nodes=top_k_nodes, ids=top_k_ids, similarities=top_k_scores
         )
+
+
+class AsyncElasticsearchStore(VectorStore):
```

Comment:
Awesome, thanks a ton for adding this!! ❤️

Quick high level comment -- I think the preferred option here would be to add async methods to the existing elastic vector store.

Checkout PGVectorStore as an example of this 👍

Reply:
Ahh cool, okay I'll give that a look; and see if I can make it more aligned to what we've got there already. 😀

Reply:
Cool! Just to further clarify, you'll want to implement `async_add` ,`adelete`, and `aquery`

The base class just calls the sync versions of those for now 
https://github.com/jerryjliu/llama_index/blob/3c427fc727eb3127cd2bcd4931f41b72c3d13ea9/llama_index/vector_stores/types.py#L147

Reply:
Awesome! thanks for this, makes sense. I'll do this tmr (its a bit late now where I'm at). ✌️🙏

Reply:
@logan-markewich: thanks again for your comments. I've tried to incorporate your suggestions in my latest push. I've updated the PR message above as well. Ultimately, I've modified `Elasticsearch` to implement async methods: `async_add`, `adelete`, and `aquery`. Hope it achieves more or less what you were hoping for here. 😀